hudi-common/src/main/java/org/apache/hudi/common/model/HoodieRecordDelegate.java

```diff
@@ -122,4 +127,20 @@ public String toString() {
         + ", newLocation=" + newLocation
         + '}';
   }
+
+  @Override
+  public final void write(Kryo kryo, Output output) {
+    kryo.writeObjectOrNull(output, hoodieKey, HoodieKey.class);
+    kryo.writeClassAndObject(output, currentLocation.isPresent() ? currentLocation.get() : null);
+    kryo.writeClassAndObject(output, newLocation.isPresent() ? newLocation.get() : null);
+  }
+
+  @Override
+  public final void read(Kryo kryo, Input input) {
+    this.hoodieKey = kryo.readObjectOrNull(input, HoodieKey.class);
+    HoodieRecordLocation newrl = (HoodieRecordLocation) kryo.readClassAndObject(input);
+    this.currentLocation = newrl == null ? Option.empty() : Option.of(newrl);
+    HoodieRecordLocation oldrl = (HoodieRecordLocation) kryo.readClassAndObject(input);
+    this.newLocation = oldrl == null ? Option.empty() : Option.of(oldrl);
+  }
```

Comment:
Option.ofNullable ? Can we write some tests for it.

Reply:
Has been updated